### ISSUE TITLE
Rule proposal: don't access `event.currentTarget` asynchronously

### ISSUE THREAD

**Issue by fregante** [sindresorhus/eslint-plugin-unicorn#2961]
### Description

Try in your console:

```js
document.body.onclick = async e=> {
    console.log('current:', e.currentTarget);
    await 1;
    console.log('current:', e.currentTarget)
}
document.body.click();

// current: body
// current: null
```

### Examples

```js
// ❌
setTimeout(() => event.currentTarget.click())

// ✅
const currentTarget = event.currentTarget;
setTimeout(() => currentTarget.click(), 1)

// ✅
setTimeout(currentTarget => currentTarget.click(), 1, event.currentTarget)
```
```js
// ❌
event.currentTarget.disabled = true
await somePromise;
event.currentTarget.disabled = false;

// ✅
event.currentTarget.disabled = true
const currentTarget = event.currentTarget;
await somePromise;
currentTarget.disabled = false;
```

```js
// ❌
event.currentTarget.href = await getUrl(location.href);

// ✅
const currentTarget = event.currentTarget;
currentTarget.href = await getUrl(location.href);
```

### Proposed rule name

no-late-current-target-access

### Additional Info

The rule should target any `event.currentTarget` access after `await` or inside nested functions. This can't catch cases where the whole `event` object is passed around, like:

```js
function handleEvent(event) {
  console.log('too late for', event.currentTarget)
}
document.body.onclick = (event) => {
  setTimeout(handleEvent, 1, event)
}
```

Related bug that could have been avoided:

- https://github.com/refined-github/refined-github/issues/9567
